src/xo-app/host/tab-network.js

```diff
@@ -117,57 +128,143 @@ class PifItem extends Component {
     }
     return reconfigurePifIp(this.props.pif, { mode })
   }
+
   _onEditIp = () => this._configIp('Static')
 
-  _editPif = vlan =>
-    editPif(this.props.pif, { vlan })
+  render () {
+    const {pif} = this.props
+    return <div>
+      {pif.ip}
+      {' '}
+      {pif.ip && <a className='hidden-md-down' onClick={this._onEditIp} style={EDIT_BUTTON_STYLE}>
+        <Icon icon='edit' size='1' fixedWidth />
+      </a>}
+    </div>
+  }
+}
+
+class PifItemMode extends Component {
+  state = { configModes: [] }
+
+  componentDidMount () {
+    getIpv4ConfigModes().then(configModes =>
+      this.setState({ configModes })
+    )
+  }
+
+  _configIp = mode => {
+    if (mode === 'Static') {
+      return confirm({
+        icon: 'ip',
+        title: _('pifConfigureIp'),
+        body: <ConfigureIpModal pif={this.props.pif} />
+      }).then(
+        params => {
+          if (!params.ip || !params.netmask) {
+            error(_('configIpErrorTitle'), _('configIpErrorMessage'))
+            return
+          }
+          return reconfigurePifIp(this.props.pif, { mode, ...params })
+        },
+        noop
+      )
+    }
+    return reconfigurePifIp(this.props.pif, { mode })
+  }
 
   render () {
-    const { networks, pif, vifsByNetwork } = this.props
+    const {pif} = this.props
     const { configModes } = this.state
+    return <Select
+      onChange={this._configIp}
+      options={configModes}
+      value={pif.mode}
+    >
+      {pif.mode}
+    </Select>
+  }
+}
 
+@connectStore(() => ({
+  vifsByNetwork: createGetObjectsOfType('VIF').groupBy('$network')
+}))
+class PifItemInUse extends Component {
+  render () {
+    const {networks, pif, vifsByNetwork} = this.props
     const pifInUse = some(vifsByNetwork[pif.$network], vif => vif.attached)
+    return <div className='text-xs-center'>
+      {_toggleDefaultLockingMode(
+        <Toggle
+          disabled={pifInUse}
+          onChange={() => editNetwork(pif.$network, { defaultIsLocked: !networks[pif.$network].defaultIsLocked })}
```

Comment:
Create this function outside the render to avoid creating a new function on each render call.